src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1312,6 +1312,44 @@ MemorySegment reinterpret(long newSize,
      */
     String getString(long offset, Charset charset);
 
+    /**
+     * Reads a string using the given byte length from this segment at the given offset,
+     * using the provided charset.
+     * <p>
+     * This method always replaces malformed-input and unmappable-character
+     * sequences with this charset's default replacement string. The {@link
+     * java.nio.charset.CharsetDecoder} class should be used when more control
+     * over the decoding process is required.
+     *
+     * @param offset  offset in bytes (relative to this segment address) at which this
+     *                access operation will occur
+     * @param length  byte length to be used for string conversion (not including any
+     *                null termination)
+     * @param charset the charset used to {@linkplain Charset#newDecoder() decode} the
+     *                string bytes
+     * @return a Java string constructed from the bytes read from the given starting
+     *         address reading the given length of characters
+     * @throws IllegalArgumentException  if the size of the string is greater than the
+     *         largest string supported by the platform
+     * @throws IndexOutOfBoundsException if {@code offset < 0}
+     * @throws IndexOutOfBoundsException if {@code offset > byteSize() - (B + N)}, where:
```

Comment:
I believe this is incorrect - e.g. all the B + N stuff is not needed, as we have a length